lib/jxl/decode.cc

```diff
@@ -1662,7 +1667,7 @@ static JxlDecoderStatus HandleBoxes(JxlDecoder* dec) {
             metadata.resize(metadata.size() * 2);
           } else if (box_result == JXL_DEC_NEED_MORE_INPUT) {
             break;  // box stage handling below will handle this instead
-          } else if (box_result == JXL_DEC_SUCCESS) {
+          } else if (box_result == JXL_DEC_BOX_COMPLETE) {
```

Comment:
shouldn't this be an or?

Reply:
As far as I can tell, it shouldn’t need to be (this specific `box_result` is the unchanged output of `JxlBoxContentDecoder::Process`, which I have changed to only ever return `BOX_COMPLETE`). But do you think it would make things clearer?

Should I revert that function to return `JXL_DEC_SUCCESS` again, and change the code above to turn it into `BOX_COMPLETE` if subscribed, rather than the other way around?